hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsClientThrottlingAnalyzer.java

```diff
@@ -95,6 +106,16 @@ private AbfsClientThrottlingAnalyzer() {
         analysisPeriodMs);
   }
 
+  public void resumeTimer() {
+    blobMetrics = new AtomicReference<AbfsOperationMetrics>(
+            new AbfsOperationMetrics(System.currentTimeMillis()));
+    timer = new Timer(String.format("abfs-timer-client-throttling-analyzer-%s", name), true);
```

Comment:
Lets not create a new timer. Reason being, timer creates a TimerThread object which never stops. This will lead to both thread-count and heap-memory to bad state.

Better is that instead of timer.cancel, do following ->
task.cancel();
timer.purge();